src/libmongoc/tests/test-mongoc-collection.c

```diff
@@ -6225,6 +6226,189 @@ test_hint_is_validated_countDocuments (void)
    mongoc_client_destroy (client);
 }
 
+#define ASSERT_INDEX_EXISTS(keys, expect_name)                               \
+   if (1) {                                                                  \
+      bool found = false;                                                    \
+      mongoc_cursor_t *cursor =                                              \
+         mongoc_collection_find_indexes_with_opts (coll, NULL /* opts */);   \
+      const bson_t *got;                                                     \
+      while (mongoc_cursor_next (cursor, &got)) {                            \
+         bson_t got_key;                                                     \
+         const char *got_name = NULL;                                        \
+         /* Results have the form: `{ v: 2, key: { x: 1 }, name: 'x_1' }` */ \
+         bsonParse (                                                         \
+            *got,                                                            \
+            require (keyWithType ("key", doc), storeDocRef (got_key)),       \
+            require (keyWithType ("name", utf8), storeStrRef (got_name)));   \
+         ASSERT_WITH_MSG (                                                   \
+            !bsonParseError, "got parse error: %s", bsonParseError);         \
+         if (bson_equal (&got_key, keys)) {                                  \
+            found = true;                                                    \
+            ASSERT_CMPSTR (got_name, expect_name);                           \
+         }                                                                   \
+      }                                                                      \
+      ASSERT_OR_PRINT (!mongoc_cursor_error (cursor, &error), error);        \
+      ASSERT_WITH_MSG (found,                                                \
+                       "expected to find index for keys: '%s', but did not", \
```

Comment:
```suggestion
                       "could not find expected index for keys: '%s'",       \
```
Avoid "but did not" being buried in the message after the keys document.